checks.d/disk.py

```diff
@@ -0,0 +1,229 @@
+            self.gauge(self.METRIC_DISK.format('write_time_pct'),
+                       write_time_pct, device_name=device_name, tags=tags)
+
+    # no psutil, let's use df
+    def collect_metrics_manually(self):
+        df_out = utils.subprocess_output.get_subprocess_output(
+            self.DF_COMMAND + ['-k'], self.log
+        )
+        self.log.debug(df_out)
+        for device in self._list_devices(df_out):
+            self.log.debug("Passed: {0}".format(device))
+            tags = [device[1]] if self._tag_by_filesystem else []
+            device_name = device[-1] if self._use_mount else device[0]
+            for metric_name, value in self._extract_metrics(device):
+                self.gauge(metric_name, value, tags=tags,
+                           device_name=device_name)
+
+    def _extract_metrics(self, device):
+        result = {}
+        # device is
+        # ["/dev/sda1", "ext4", 524288,  171642,  352646, "33%", "/"]
+        result[self.METRIC_DISK.format('total')] = float(device[2])
+        result[self.METRIC_DISK.format('used')] = float(device[3])
+        result[self.METRIC_DISK.format('free')] = float(device[4])
+        if len(device[5]) > 1 and device[5][-1] == '%':
+            result[self.METRIC_DISK.format('in_use')] = \
+                float(device[5][:-1]) / 100.0
+
+        result.update(self._collect_inodes_metrics(device[-1]))
+        return result.iteritems()
+
+    def _keep_device(self, device):
+        # device is for Unix
+        # [/dev/disk0s2, ext4, 244277768, 88767396, 155254372, 37%, /]
+        # First, skip empty lines.
+        # then filter our fake hosts like 'map -hosts'.
+        #    Filesystem    Type   1024-blocks     Used Available Capacity  Mounted on
+        #    /dev/disk0s2  ext4     244277768 88767396 155254372    37%    /
+        #    map -hosts    tmpfs            0        0         0   100%    /net
+        # and finally filter out fake devices
+        return (device and len(device) > 1 and
+                device[2].isdigit() and
+                not self._exclude_disk(device[0], device[1]))
+
+    def _flatten_devices(self, devices):
+        # Some volumes are stored on their own line. Rejoin them here.
+        previous = None
+        for parts in devices:
+            if len(parts) == 1:
+                previous = parts[0]
+            elif previous and self._is_number(parts[0]):
+                # collate with previous line
+                parts.insert(0, previous)
+                previous = None
+            else:
+                previous = None
+        return devices
+
+    def _list_devices(self, df_output):
+        """
+        Given raw output for the df command, transform it into a normalized
+        list devices. A 'device' is a list with fields corresponding to the
+        output of df output on each platform.
+        """
+        all_devices = [l.strip().split() for l in df_output.split("\n")]
```

Comment:
I know it doesn't matter in this case as it's only for linux but you should use:
```python
df_output.splitlines()
```
instead. It's better practice as it should work on all platforms.